qt/widgets/common/src/pqHelpWindow.cxx

```diff
@@ -369,15 +369,15 @@ void pqHelpWindow::showLinkedPage(const QUrl &url) { this->showPage(url, true);
 
 //-----------------------------------------------------------------------------
 void pqHelpWindow::printPage() {
-  QPrinter printer;
-  QPrintDialog dialog(&printer, this);
+  auto *printer = new QPrinter();
```

Comment:
Nice catch!

Does this create a memory leak or does the dialog take ownership?

Reply:
Likely to be a memory leak, unless a `QPrinter` is smart enough to destruct once printing is completed - I can't find any documentation to this regard.

I'm not sure how we can fix this otherwise - the `QPrinter` is still needed after this function returns.

Reply:
🤔 
https://doc.qt.io/qt-5/qwebenginepage.html#print says that "It is the users responsibility to ensure the printer remains valid until resultCallback has been called". I wonder if we can do:

```c++
m_browser->page()->print(printer, [printer](bool) {printer->deleteLater();});
```
? I think this will allow the callback to complete and then delete the QPrinter on the next trip around the Qt event loop.

We don't use the webkit any more now that we are on Qt 5.15 so that code is obselete now and thankfully we don't need to worry about it.

Reply:
Sounds promising - I will implement and test